runtime/src/main/scala/akka/grpc/scaladsl/headers/headers.scala

```diff
@@ -81,19 +87,86 @@ object `Status` extends ModeledCustomHeaderCompanion[`Status`] {
     headers.collectFirst { case h if h.is(name) => Integer.parseInt(h.value()) }
 }
 
-// TODO percent-encoding of message?
-final class `Status-Message`(override val value: String) extends ModeledCustomHeader[`Status-Message`] {
+// grpc-message must be percent encoded: https://github.com/grpc/grpc/issues/4672
+final class `Status-Message`(val unencodedValue: String) extends ModeledCustomHeader[`Status-Message`] {
+  import `Status-Message`._
+
   override def renderInRequests = false
   override def renderInResponses = true
   override val companion = `Status-Message`
+  override def value() = toAsciiString(unencodedValue)
+
+  // Copied with slight adaptations from https://github.com/grpc/grpc-java/blob/79e75bace40cea7e4be72e7dcd1f41c3ad6ee857/api/src/main/java/io/grpc/Status.java#L582
+  private def toAsciiString(value: String): String = {
+    val valueBytes = value.getBytes(StandardCharsets.UTF_8)
+    val firstIndexToEscape = valueBytes.indexWhere(isEscapingChar(_))
+    if (firstIndexToEscape == -1)
+      new String(valueBytes)
```

Comment:
return `value` instead of re-wrapping inside a new String?

Reply:
I know it seems like re-wrapping the same content but actually it's not

The following is not always true for any `value`: `new String(value.getBytes(StandardCharsets.UTF_8)) == value`. Consider `"a \ud801"`, its underlying array of bytes is `[97, 0, 32, 0, 1, -40]`. It's a UTF-16 string, note that even 'a' is encoded with 2 bytes. `"a \ud801".getBytes(StandardCharsets.UTF_8)` yields `[97, 32, 63]`, there are 2 effects: ascii characters are encoded with 1 byte and lone surrogate has been replaced with `?` (63 in ascii)